### PR TITLE
Removes BatchingPageCache (backport from 2.3)

### PR DESCRIPTION
**Pull request by tinwelint**
in favor of plain MuninnPageCache. This to reduce code complexity and the
burden to maintain two implementations of PageCache.

Observed throughput is slightly lower in MuninnPageCache due to the
shortcuts taken in the BatchingPageCache for the specific use case of the
batch importer, but that matters quite little since page cache I/O wasn't
and doesn't seem to have a likeliness to be the bottleneck.

In short the batch import performance isn't affected by this.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/io/IoTracer.java [hunk 1]
```diff
@@ -20,17 +20,18 @@
...
-    void dataWritten( int bytes );
+    long countBytesWrittem();
 
-    public static final Monitor NO_MONITOR = new Monitor()
+    public static final IoTracer NONE = new IoTracer()
     {
         @Override
-        public void dataWritten( int bytes )
-        {   // Do nothing
+        public long countBytesWrittem()
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/io/IoTracer.java:32]: `countBytesWrittem`->`countBytesWritten`?

Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/io/IoTracer.java:32]: or a pun on `countBytesRhythm`? good catch, will change

**Comment by tinwelint** [comment]
Closing since unnecessary change
